templates/generate.py

```diff
@@ -0,0 +1,121 @@
+import argparse
+import json
+import logging
+import os
+import shutil
+import tempfile
+from pathlib import Path
+from typing import Any, Dict, List
+
+import yaml
+from pydantic import BaseModel
+from truss.patch.hash import directory_content_hash
+
+
+class Replacement(BaseModel):
+    from_str: str
+    to_str: str
+
+
+class Generate(BaseModel):
+    based_on: str
+    config: Dict[str, Any]
+    ignore: List[str]
+    replaces: Dict[str, Replacement]
+
+
+def process(dst: Path, templates: Path, generate: Generate, only_check: bool):
+    logging.info(f"processing {str(dst)}")
+    with tempfile.TemporaryDirectory() as temp_dir:
+        # copy template
+        for filename in os.listdir(templates / generate.based_on):
+            source_file = templates / generate.based_on / filename
+            destination_file = Path(temp_dir) / filename
+
+            if os.path.isfile(source_file):
```

Comment:
nit: best to use `pathlib.Path` functions over `os.path`. Is this case `source_file.is_file()`. Later you can use `source_file.read_text/write_text` etc